content/blog/recipes-1-1-0/index.Rmd

```diff
@@ -0,0 +1,239 @@
+
+```{r}
+#| error: true
+# didn't use to throw a warning
+recipe(mtcars) |>
+  update_role(everything(), new_role = "predictor") |>
+  add_role("mpg", new_role = "outcome")
+```
+
+This specific problem can be dealt with using `update_role()` instead of `add_role()`.
+
+```{r}
+recipe(mtcars) |>
+  update_role(everything(), new_role = "predictor") |>
+  update_role("mpg", new_role = "outcome")
+```
+
+## Long formulas in recipe()
+
+Related to the changes we saw above, we now fully support very long formulas without hitting a `C stack usage` error.
+
+```{r}
+data_wide <- matrix(1:10000, ncol = 10000)
+data_wide <- as.data.frame(data_wide)
+names(data_wide) <- c(paste0("x", 1:10000))
+
+long_formula <- as.formula(paste("~ ", paste(names(data_wide), collapse = " + ")))
+
+recipe(long_formula, data_wide)
+```
+
+## Better error for misspelled argument names
+
+If you have used recipes long enough you are very likely to have run into the following error:
+
+``` r
+recipe(mpg ~ ., data = mtcars) |>
+  step_pca(all_numeric_predictors(), number = 4) |>
+  prep()
+#> Error in `step_pca()`:
+#> Caused by error in `prep()`:
+#> ! Can't rename variables in this context.
+```
+
+and the first time you saw it, it didn't make much sense. Hopefully, you figured out that [step_pca()](https://recipes.tidymodels.org/reference/step_pca.html) doesn't have a `number` argument, and instead uses `num_comp` to determine the number of principal components to return. This confusion will be a thing of the past as we now include this improved error message:
+
+```{r}
+#| error: true
+recipe(mpg ~ ., data = mtcars) |>
+  step_pca(all_numeric_predictors(), number = 4) |>
+  prep()
+```
+
+## Quality of life increases in step_dummy()
+
+I would imagine that one of the most used steps is `step_dummy()`. We have improved the errors and warnings it spits out when things go sideways.
+
+If you apply `step_dummy()` to a variable that contains a lot of levels, it will produce a lot of columns, which depending on the size of your data won't fit in memory. This can lead to the following error:
+
+```r
+data_id <- tibble(
+  id = as.character(1:100000), 
+  x1 = rnorm(100000), 
+  x2 = sample(letters, 100000, TRUE)
+)
+
+recipe(~ ., data = data_id) |>
+  step_dummy(all_nominal_predictors()) |>
+  prep()
+#> Error: vector memory exhausted (limit reached?)
+```
+
+Instead, you now get a more helpful error message.
+
+```{r}
+#| error: true
+data_id <- tibble(
+  id = as.character(1:100000), 
+  x1 = rnorm(100000), 
+  x2 = sample(letters, 100000, TRUE)
+)
+
+recipe(~ ., data = data_id) |>
+  step_dummy(all_nominal_predictors()) |>
+  prep()
+```
+
+Likewise, you will get helpful errors if `step_dummy()` gets a `NA` or unseen values
+
+```{r}
+data_train <- tibble(x = c("a", "b"))
+data_unseen <- tibble(x = "c")
+
+rec_spec <- recipe(~., data = data_train) %>%
+  step_dummy(x) %>%
+  prep()
+
+rec_spec %>%
+  bake(data_unseen)
```

Comment:
The resulting warning has a stray `\` in it, before the word `before`: 

> #> Warning: ! There are new levels in `x`: "c".
#> ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
#>   `step_dummy()` to handle unseen values.

Reply:
fixed in https://github.com/tidymodels/recipes/pull/1346

cheating and rendering with that version to avoid this